source/opt/const_folding_rules.cpp

```diff
@@ -272,65 +272,63 @@ ConstantFoldingRule FoldFPBinaryOp(BinaryScalarFoldingRule scalar_rule) {
 // This macro defines a |UnaryScalarFoldingRule| that performs float to
 // integer conversion.
 // TODO(greg-lunarg): Support for 64-bit integer types.
-#define FOLD_FTOI_OP()                                                        \
-  [](const analysis::Type* result_type, const analysis::Constant* a,          \
-     analysis::ConstantManager* const_mgr) -> const analysis::Constant* {     \
-    assert(result_type != nullptr && a != nullptr);                           \
-    const analysis::Integer* integer_type = result_type->AsInteger();         \
-    const analysis::Float* float_type = a->type()->AsFloat();                 \
-    assert(float_type != nullptr);                                            \
-    assert(integer_type != nullptr);                                          \
-    if (integer_type->width() != 32)                                          \
-      return nullptr;                                                         \
-    if (float_type->width() == 32) {                                          \
-      float fa = a->GetFloat();                                               \
-      uint32_t result = integer_type->IsSigned()                              \
-                            ? static_cast<uint32_t>(static_cast<int32_t>(fa)) \
-                            : static_cast<uint32_t>(fa);                      \
-      std::vector<uint32_t> words = {result};                                 \
-      return const_mgr->GetConstant(result_type, words);                      \
-    } else if (float_type->width() == 64) {                                   \
-      double fa = a->GetDouble();                                             \
-      uint32_t result = integer_type->IsSigned()                              \
-                            ? static_cast<uint32_t>(static_cast<int32_t>(fa)) \
-                            : static_cast<uint32_t>(fa);                      \
-      std::vector<uint32_t> words = {result};                                 \
-      return const_mgr->GetConstant(result_type, words);                      \
-    }                                                                         \
-    return nullptr;                                                           \
-  }
+UnaryScalarFoldingRule FoldFToIOp =
```

Comment:
Small problem.  This is technically a global that is not a trivially destructible (I think). https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables